api/v1beta2/awsmachine_webhook.go

```diff
@@ -334,6 +336,29 @@ func (r *AWSMachine) validateRootVolume() field.ErrorList {
 	return allErrs
 }
 
+func (r *AWSMachine) validateNetworkElasticIPPool() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if r.Spec.ElasticIPPool == nil {
+		return allErrs
+	}
+	if !ptr.Deref(r.Spec.PublicIP, false) {
+		allErrs = append(allErrs, field.Required(field.NewPath("spec.elasticIpPool"), "publicIp must be set to 'true' to assign custom public IPv4 pools with elasticIpPool"))
+	}
+	eipp := r.Spec.ElasticIPPool
+	if eipp.PublicIpv4Pool != nil {
+		if eipp.PublicIpv4PoolFallBackOrder == nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.elasticIpPool.publicIpv4PoolFallbackOrder"), r.Spec.ElasticIPPool, "publicIpv4PoolFallbackOrder must be set when publicIpv4Pool is defined."))
+		}
+		awsPublicIpv4PoolPrefix := "ipv4pool-ec2-"
+		if !strings.HasPrefix(*eipp.PublicIpv4Pool, awsPublicIpv4PoolPrefix) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.elasticIpPool.publicIpv4Pool"), r.Spec.ElasticIPPool, fmt.Sprintf("publicIpv4Pool must start with %s.", awsPublicIpv4PoolPrefix)))
+		}
+	}
```

Comment:
Do we need the `eipp.PublicIpv4Pool == nil && eipp.PublicIpv4PoolFallBackOrder != nil` case here?

Reply:
hey @nrb - the validation `validateNetworkElasticIPPool()` must cover the scenarios bellow, which includes your suggestion. Am I missing something?

| PublicIpv4Pool | PublicIpv4PoolFallBackOrder | covered line |
| -- | -- | -- |
| set (valid) | set | pass L361 |
| set | nil  | err L351 |
| invalid value | set | err L355 |
| nil | set | err L358 |
| nil  | nil | pass L361 |

Reply:
Ah, I see what I missed - I read L357 as being nested within the check at L349. Thanks for the clarification.